packages/lab/src/stepper-input/internal/useActivateWhileMouseDown.ts

```diff
@@ -0,0 +1,49 @@
+import { type SyntheticEvent, useCallback, useEffect, useState } from "react";
+import { useInterval } from "./useInterval";
+
+const INITIAL_DELAY = 500;
+const INTERVAL_DELAY = 100;
+
+export function useActivateWhileMouseDown(
+  activationFn: (event?: SyntheticEvent) => void,
+  isAtLimit: boolean,
+) {
+  const [buttonDown, setButtonDown] = useState(false);
+  const [delay, setDelay] = useState(INITIAL_DELAY);
+
+  const cancelInterval = useCallback(() => {
+    setButtonDown(false);
+    setDelay(INITIAL_DELAY);
+  }, []);
+
+  useEffect(() => {
+    if (isAtLimit) cancelInterval();
+  }, [isAtLimit, cancelInterval]);
+
+  useEffect(() => {
+    window.addEventListener("mouseup", cancelInterval);
```

Comment:
should this be reading window from useWindow ?

Reply:
yes, good catch